### PR TITLE
[refactor] Use Elisp style, `-p` predicates

### PR DESCRIPTION
**Pull request by benedekfazekas**
As discussed in clojure-emacs/clj-refactor.el#311 `-p` style predicates
are more consistent with the rest of clojure-emacs projects and the
language in general.

cljr predicates renamed

- cljr--use-multiple-cursors?
- cljr--comment-line?
- cljr--whitespace?
- cljr--inside-prefixed-libspec-vector?
- cljr--even?
- cljr--op-supported?
- cljr--cljs-file?
- cljr--cljc-file?
- cljr--clj-file?
- cljr--dash-in-file-name?
- cljr--already-declared?
- cljr--existing-group-of-defs?
- cljr--point-in-reader-conditional?
- cljr--point-in-reader-conditional-branch?
- cljr--clj-context?
- cljr--in-namespace-declaration?
- cljr--excluded-from-project-clean?
- cljr--empty-buffer?
- cljr--end-of-buffer?
- cljr--qualified-symbol?
- cljr--is-keyword?
- cljr--symbol?
- cljr--keyword-lookup?
- cljr--defn?
- cljr--call-site?
- cljr--no-changes-to-parameter-order?
- cljr--apply-call-site?
- cljr--partial-call-site?
- cljr--ignorable-occurrence?

predicates renamed from dependencies

- -contains? (dash)
- -every? (dash)
- s-contains? (s)
- s-starts-with? (s)
- s-ends-with? (s)
- s-equals (s)
- s-matches? (s)

predicates not renamed as there is no suitable alias

- s-blank? (s)
- s-present? (s)
- s-uppercase? (s)

### CODE REVIEW

**Comment by bbatsov** [comment]
The changes look good to me. And the last section of the commit is a reminder that it's better to use seq and subr-x in the long run (for pretty much the same reasons).

File: clj-refactor.el [hunk 1]
```diff
@@ -724,7 +724,7 @@ at the opening parentheses of an anonymous function."
           (when (<= (point) search-bound)
             (error "Can't find definition of anonymous function!")))))))
 
-(defun cljr--even? (n)
+(defun cljr--even-p (n)
```
Comment (expez) [inline clj-refactor.el]: This should've been `cljr--evenp`.  There's only a hyphen before the `p` if there's a hyphen in the predicate.

See more [here](https://www.cs.cmu.edu/Groups/AI/html/cltl/clm/node69.html) if you're **really** interested.

File: clj-refactor.el [hunk 2]
```diff
@@ -2171,13 +2171,13 @@ FEATURE is either :clj or :cljs."
 
 (defun cljr--get-aliases-from-middleware ()
   (-when-let (aliases (cljr--call-middleware-for-namespace-aliases))
-    (if (cljr--clj-context?)
+    (if (cljr--clj-context-p)
         (gethash :clj aliases)
       (gethash :cljs aliases))))
 
 (defun cljr--is-global-alias (alias)
```
Comment (expez) [inline clj-refactor.el]: `cljr--global-alias-p`, though I'd prefer we rename this altogether.  When reading the code it wasn't at all obvious what it was until I saw the implementation.  Perhaps `cljr--cljs-js-alias-p` or just `cljr--js-alias-p`?

File: clj-refactor.el [hunk 3]
```diff
@@ -3635,22 +3635,22 @@ See: https://github.com/clojure-emacs/clj-refactor.el/wiki/cljr-create-fn-from-e
...
-  (if (s-starts-with? "(->" s)
+  (if (s-starts-with-p "(->" s)
       (cljr--with-string-content s
         (cljr-unwind-all)
         (buffer-substring (point-min) (point-max)))
     s))
 
-(defun cljr--is-keyword? (s)
-  (s-matches? "^:[^0-9:[{(\"][^[{(\"]*$"
+(defun cljr--keyword-p (s)
```
Comment (expez) [inline clj-refactor.el]: `cljr--keywordp`

File: clj-refactor.el [hunk 4]
```diff
@@ -3635,22 +3635,22 @@ See: https://github.com/clojure-emacs/clj-refactor.el/wiki/cljr-create-fn-from-e
...
+  (s-matches-p "^:[^0-9:[{(\"][^[{(\"]*$"
               (s-replace "\n" " " s)))
 
-(defun cljr--symbol? (s)
+(defun cljr--symbol-p (s)
```
Comment (expez) [inline clj-refactor.el]: `cljr--symbolp`

File: clj-refactor.el [hunk 5]
```diff
@@ -3914,8 +3914,8 @@ See: https://github.com/clojure-emacs/clj-refactor.el/wiki/cljr-describe-refacto
     (insert new-name)
     (view-mode 1)))
 
-(defun cljr--defn? (match)
-  (s-matches? (rx (seq line-start (* whitespace) "("
+(defun cljr--defn-p (match)
```
Comment (expez) [inline clj-refactor.el]: `cljr--defnp`

**Comment by expez** [comment]
Great work @benedekfazekas!

File: clj-refactor.el (hunk 1 above)
Comment (bbatsov) [inline clj-refactor.el]: Ah, yes. Even I forgot about this.

**Pull request merged**